gammapy/scripts/tests/test_analysis.py

```diff
@@ -21,3 +22,79 @@ def test_validate_astropy_quantities():
     config = {"observations": {"filter": [{"lon": "1 deg"}]}}
     analysis = Analysis(config)
     assert analysis.config.validate() is None
+
+
+@pytest.mark.parametrize(
+    "config",
+    [
+        {
+            "observations": {
+                "datastore": "$GAMMAPY_DATA/cta-1dc/index/gps/",
+                "filter": [{"filter_type": "all"}],
+            },
+            "result": 4},
+        {
+            "observations": {
+                "datastore": "$GAMMAPY_DATA/cta-1dc/index/gps/",
+                "filter": [{"filter_type": "ids", "obs_ids": [110380]}],
+            },
+            "result": 1,
+        },
+        {
+            "observations": {
+                "datastore": "$GAMMAPY_DATA/cta-1dc/index/gps/",
+                "filter": [
+                    {"filter_type": "all"},
+                    {"filter_type": "ids", "obs_ids": [110380], "exclude": True},
+                ]
+            },
+            "result": 3,
+        },
+        {
+            "observations": {
+                "datastore": "$GAMMAPY_DATA/cta-1dc/index/gps/",
+                "filter": [
+                    {
+                        "filter_type": "sky_circle",
+                        "frame": "galactic",
+                        "lon": "0 deg",
+                        "lat": "0 deg",
+                        "radius": "1 deg",
+                        "border": "0.5 deg",
+                    }
+                ]
+            },
+            "result": 1,
+        },
+        {
+            "observations": {
+                "datastore": "$GAMMAPY_DATA/cta-1dc/index/gps/",
+                "filter": [
+                    {
+                        "filter_type": "angle_box",
+                        "variable": "RA_PNT",
+                        "value_range": ["265 deg", "268 deg"],
+                    }
+                ]
+            },
+            "result": 2,
+        },
+        {
+            "observations": {
+                "datastore": "$GAMMAPY_DATA/cta-1dc/index/gps/",
+                "filter": [
+                    {
+                        "filter_type": "par_box",
+                        "variable": "EVENT_COUNT",
+                        "value_range": [106000, 107000],
+                    }
+                ]
+            },
+            "result": 2,
+        },
+    ],
+)
```

Comment:
I wonder if we should start using YAML multi-line text in Gammapy tests instead of dicts, because it's a bit easier to read and write and shorter?
```
    config = """
    observations:
        datastore: ...
        filter: ...
    """
```
@Bultako - Better or worse than the current dicts?

Reply:
Yes! I take it.
Do you have an example on how to do it?

Reply:
Can you try to copy & paste your dict from the test Python file, and use `yaml.dump` to get the corresponding string? Like I said, I think this will be more readable and shorter for config, but I'm not 100% sure.

Reply:
Ok, I've done this.
I think it is more readable now.
https://github.com/gammapy/gammapy/blob/d26e574cb7a32469032cca74fc9a9f3894b19ec1/gammapy/scripts/tests/test_analysis.py#L28-L86

Reply:
Agreed, it's better.

I think we'll want a mini language, e.g. for regions we use DS9 strings, and for other things we use strings like "zenith > 10" and parse them.

But this really isn't high-priority, let's leave as-is and come back to obs selection later.